File: app/src/main/java/org/autojs/autojs/core/ui/JsViewHelper.java
package org.autojs.autojs.core.ui;

import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;

import org.autojs.autojs.core.ui.inflater.util.Ids;

/**
 * Created by Stardust on 2017/5/14.
 */
public class JsViewHelper {

    @Nullable
    public static View findViewByStringId(View view, String id) {
        View result = view.findViewById(Ids.parse(id));
        if (result != null)
            return result;
        if (!(view instanceof ViewGroup group)) {
            return null;
        }
        for (int i = 0; i < group.getChildCount(); i++) {
            result = findViewByStringId(group.getChildAt(i), id);
            if (result != null)
                return result;
        }
        return null;
    }

}

File: app/src/main/java/org/autojs/autojs/core/ui/JsViewHelper.kt
package org.autojs.autojs.core.ui

import android.view.View
import android.view.ViewGroup
import org.autojs.autojs.core.ui.inflater.util.Ids

/**
 * Created by Stardust on May 14, 2017.
 * Transformed by 抠脚本人 (https://github.com/little-alei) on Jul 10, 2023.
 */
object JsViewHelper {
    @JvmStatic
    fun findViewByStringId(view: View, id: String?): View? {
        view.findViewById<View>(Ids.parse(id))?.let { return it }
        if (view is ViewGroup) {
            for (i in 0 until view.childCount) {
                findViewByStringId(view.getChildAt(i), id)?.let { return it }
            }
        }
        return null
    }
}
